adhocteam/Head-Start-TTADP
# Terraform Init

**Description of change**
Create persistent infrastructure with Terraform.

**Reasoning behind RDS plan choices:** 
- `micro-psql` for dev and staging: smallest dedicated instance that allows us to use postgres v12.4
- `small-psql` for prod: creates a small-t3 instance, 1vCPU and 2 GiB RAM should be sufficient for now. We'll be better able to tune this to the appropriate size when we have prod deployed and a monitoring system set up. Performing an update to the instance size will cause only a brief period of downtime (second to minutes). A developer can easily and autonomously scale an instance to a larger plan, however, scaling an instance to a smaller plan requires us to email cloud.gov support.

**Note to reviewers:** This is infrastructure configuration, not code.  It is highly repetitive and not modular by design. The terraform block must be repeated for every environment and parameters for those tags do not accept variables as input.

**How to test**
- Follow the extensive directions (or just sanity check them) at terraform/README.md
- @jasalisbury agreed to be my tester and he worked through the directions at terraform/README.md to create a rds instance for dev. You can [check out the RDS instance in cloud.gov](https://dashboard.fr.cloud.gov/services).

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/23

**Checklist**

- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [x] Documentation updated - **added new section to main README, created terraform/README.md**
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
